tests/test_executor/test_executors.py

```diff
@@ -13,63 +13,93 @@
 
 
 class TestExecutor(TestEGCG):
+    executor_cls = Executor
+
+    def execute(self, *args, **kwargs):
+        e = self.executor_cls(*args, **kwargs)
+        try:
+            e.start()
+        except NotImplementedError:
+            pass
+
+        return e.join()
+
     def test_cmd(self):
-        e = Executor('ls ' + os.path.join(self.assets_path, '..'))
-        exit_status = e.join()
-        assert exit_status == 0
+        assert self.execute('ls ' + os.path.dirname(self.assets_path)) == 0
 
     def test_dodgy_cmd(self):
-        with pytest.raises(EGCGError) as err:
-            e = Executor('dodgy_cmd')
-            e.join()
-            assert 'Command failed: \'dodgy_cmd\'' in str(err)
+        assert self.execute('dodgy_cmd') == 127
+
+    def test_script(self):
+        assert self.execute(os.path.join(self.assets_path, 'countdown.sh')) == 0
+
+    def test_dodgy_script(self):
+        # same exit status as the running script
+        assert self.execute('%s dodgy' % os.path.join(self.assets_path, 'countdown.sh')) == 13
```

Comment:
@mwhamgenomics Same here, it would be helpful to include what the error numbers actually mean.

Reply:
Same as above, except this is testing the return value of countdown.sh